### ISSUE TITLE
Picture update

### ISSUE THREAD

**Issue by JTrotta** [nopSolutions/nopCommerce#5564]
nopCommerce version: latest

Steps to reproduce the problem: 

1. set media storage to file system
2. insert picture
3. file picture in /images is empty

I suppose there is a bug here:

```
        public virtual async Task<Picture> UpdatePictureAsync(Picture picture)
        {
            if (picture == null)
                return null;
            var seoFilename = CommonHelper.EnsureMaximumLength(picture.SeoFilename, 100);

            //delete old thumbs if a picture has been changed
            if (seoFilename != picture.SeoFilename)
                await DeletePictureThumbsAsync(picture);

            picture.SeoFilename = seoFilename;

            await _pictureRepository.UpdateAsync(picture);
            await UpdatePictureBinaryAsync(picture, await IsStoreInDbAsync() ? (await GetPictureBinaryByPictureIdAsync(picture.Id)).BinaryData : Array.Empty<byte>());

            if (!await IsStoreInDbAsync())
                await SavePictureInFileAsync(picture.Id, (await GetPictureBinaryByPictureIdAsync(picture.Id)).BinaryData, picture.MimeType);

            return picture;
        }
```

if IsStoreInDbAsync is false  `GetPictureBinaryByPictureIdAsync(picture.Id)).BinaryData` is always empty and this line:
`await SavePictureInFileAsync(picture.Id, (await GetPictureBinaryByPictureIdAsync(picture.Id)).BinaryData, picture.MimeType);
`
overwrites file with empty array.

**Comment by firoz3bits** [comment]
I also face the same issue, after switching to filesystem.

the work around is to comment the following lines, as it makes no sense to replace the binary.

```
await UpdatePictureBinaryAsync(picture, await IsStoreInDbAsync() ? (await GetPictureBinaryByPictureIdAsync(picture.Id)).BinaryData : Array.Empty<byte>());

if (!await IsStoreInDbAsync())
                await SavePictureInFileAsync(picture.Id, (await GetPictureBinaryByPictureIdAsync(picture.Id)).BinaryData, picture.MimeType);

```

When files are stored in filesystem, the binary data is empty

![image](https://user-images.githubusercontent.com/71256628/133431314-8eaed175-a74c-4b4f-837b-ce555561ad11.png)

so this method, it fetches this empty data and replaces the image with this empty data.
Hence this issue.

I have not fully tested if commenting out these line would cause any other issues.

**Comment by holydk** [comment]
@JTrotta @firoz3bits I can't reproduce issue on develop branch.
The steps I followed
1. Installed a fresh develop
2. Switched to storing pictures in the file system
3. Inserted pictures on the product page. Added one to the main pictures and one to the description.
4. Check public. Works fine
5. Then I switched several times to storing in the database and file system

Am I missing something? Can you confirm that the issue is being reproduced on the develop branch?

**Comment by ahmadkq** [comment]
I am facing the same issues. Many of pictures saved with 0 size to the thumb folder!

**Comment by skoshelev** [comment]
I checked twice, everything works as expected, maybe some kind of glitch when changing the image storage location

Closed #5564
